lib/utils/visUtil.cpp

```diff
@@ -301,14 +308,36 @@ double StatTracker::get_max() {
     if (count == 0) {
         return NAN;
     }
-    return min_max.get_max();
+
+    if (is_optimized) {
+        return min_max.get_max();
+    } else {
+        // brute force way to get max
+        double max = std::numeric_limits<double>::lowest();
+        int start = (end - count) % buf_size;
+        for (size_t i = 0; i < count; i++) {
+            max = std::max(max, rbuf[(start + i) % buf_size].value);
+        }
```

Comment:
The logic here is more complex than it needs to be.   In the case that `count < buf_size` since we always start at `0` you can just start with `i = 0`, and once `count == buf_size` then we only need to find the max of the entire array - it doesn't matter where we start.   So a simple `rbuf[i].value` will do the trick.  Same for `min`.